### ISSUE TITLE
Can't compile in Docker after PR #2845

### ISSUE THREAD

**Issue by vsky279** [nodemcu/nodemcu-firmware#2853]
### Expected behavior
After PR #2845 and PR #2844 (probably only one of them being toxic) I am not able to compile `dev` version with Docker (Windows 10).

### Actual behavior
```
CC app/user/user_main.c
In file included from user_main.c:25:0:
../../sdk-overrides/include/espconn.h:1:1: error: expected identifier or '(' before '.' token
 ../../app/include/lwip/app/espconn.h
 ^
user_main.c: In function 'user_process_input':
user_main.c:298:32: error: 'input_sig' undeclared (first use in this function)
     return task_post_low(input_sig, force);
                                ^
user_main.c:298:32: note: each undeclared identifier is reported only once for each function it appears in
user_main.c: In function 'user_init':
user_main.c:340:5: error: 'input_sig' undeclared (first use in this function)
     input_sig = task_get_id(handle_input);
     ^
make[2]: *** [.output/eagle/debug/obj/user_main.o] Error 1
make[2]: Leaving directory `/opt/nodemcu-firmware/app/user'
make[1]: *** [.subdirs] Error 2
make[1]: Leaving directory `/opt/nodemcu-firmware/app'
make: *** [.subdirs] Error 2
make: Leaving directory `/opt/nodemcu-firmware'
```

### Test code
No test code is needed. I just can't compile clean pull of `dev` version with the error above.

### NodeMCU version
`dev`

### Hardware
N/A

**Comment by HHHartmann** [comment]
Can confirm

**Comment by marcelstoer** [comment]
> PR #2844

How is that one related? It only changed documentation for the ESP32 branch.

**Comment by marcelstoer** [comment]
@HHHartmann since the NodeMCU CI build is obviously ok do you think this may be related to trusty vs xenial (and therefore related to https://github.com/marcelstoer/docker-nodemcu-build/pull/69)?

**Comment by vsky279** [comment]
> > PR #2844
> 
> How is that one related? It only changed documentation for the ESP32 branch.

Then it is not related. Of course. 
Sorry I did not go into detail. I just saw that after these two PRs compile fails.

**Comment by HHHartmann** [comment]
seems to be that sdk-overrides/include/espconn.h is a link which does not get checked correctly out under windows as mentioned elsewhere. It contains 
```
../../app/include/lwip/app/espconn.h
```

**Comment by HHHartmann** [comment]
After copying `app/include/lwip/app/espconn.h` to `sdk-overrides/include/espconn.h` it compiles.

**Comment by marcelstoer** [comment]
> not get checked correctly out under windows

Ah, that explains why I cannot reproduce this on macOS.

**Comment by HHHartmann** [comment]
Fixed on dev now

**Comment by marcelstoer** [comment]
This was a bit of tumbleweed. The actual culprit was https://github.com/nodemcu/nodemcu-firmware/commit/98c2c0520d221d9fbbb2107d7273cd64af3535fd#diff-08924738929fb5bc35be7f4839a1f1c5 from @TerryE merged in #2843. 

Sorry I forgot to close this issue after I merged @jmattsson's fix.
